### PR TITLE
feature/phase 4

### PR DESCRIPTION
**Pull request by zubairaltaf10**

### CODE REVIEW

File: src/Application/Events/UserCreateEvent.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import EventBus from 'events'
+
+class UserCreatedEvent extends EventBus {
+    constructor(){
+        super()
+        this.handleEvent()
+    }
+
+    protected async handleEvent() {
+        this.on('userCreated',((user)=>{
```
Comment (bilalfaisal-ct) [inline src/Application/Events/UserCreateEvent.ts:11]: You can put all events in constants.

File: src/Application/User/UserService.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    constructor() {
+        super()
+
+    }
+
+    async createUser(res: Response, req: Request, user: User) {
+        try {
+            const userEntity = UserEntity.create(user)
+            const createUserEntity = new CreateUserCommand(userEntity);
+            var result = await commandBus.handle(createUserEntity);
```
Comment (bilalfaisal-ct) [inline src/Application/User/UserService.ts:26]: Instead of var it can be const.

File: src/Application/User/UserService.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            if (result) {
```
Comment (bilalfaisal-ct) [inline src/Application/User/UserService.ts:27]: I think the it should return if (!result) return;

and the other return statement can go as it is.

File: src/infrastructure/ThirdPartyClients/Email/EmailService.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  constructor() {
+    this.connection();
+  }
+
+  private connection() {
+    this.transporter = nodemailer.createTransport({
+      service: 'gmail',
+      auth: {
+        user: 'zubair.altaf@carbonteq.com',
```
Comment (bilalfaisal-ct) [inline src/infrastructure/ThirdPartyClients/Email/EmailService.ts:14]: These can be loaded from configs.

**Review by bilalfaisal-ct — CHANGES_REQUESTED** [review]
Please see comments, Thanks

File: Http/Controllers/UserController.ts [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
-import  userService  from "../../src/services/userService";
+import GoogleAuthService from '../../src/Application/Auth/GoogleAuthService';
+import  userService  from "../../src/Application/User/UserService";
 
 
 class UserController {
-    register(req: Request, res: Response, next: NextFunction) {
-        return userService.createUser( res,req,req.body)    
+
+    Register(req: Request, res: Response, next: NextFunction) {
```
Comment (bilalfaisal-ct) [inline Http/Controllers/UserController.ts:8]: i think the methods can be renamed to camelCase

File: Http/Controllers/TodoController.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { UserModel } from "../../src/Infrastructure/Database/Models/User"
+import { Request, Response, NextFunction } from 'express';
+import TodoService from "../../src/Application/Todo/TodoService";
+
+class TodoController {
+   
+    async createTODO(req: Request, res: Response, next: NextFunction) {
+
+       return TodoService.CreateTodo(res,req.body)
```
Comment (zubairaltaf10) [inline Http/Controllers/TodoController.ts:9]: camel case for methods

**Pull request closed without merging**